src/entities/modal-window/components/ModalWindow.vue

```diff
@@ -0,0 +1,43 @@
+<script setup lang="ts">
+import { ref } from 'vue'
+
+import { modal, closeModal, open, catchFocus } from '../lib/modal'
+
+const button = ref(null)
```

Comment:
На текущий момент твоя проблема с типами решает вот так:

```suggestion
import { createElementRef } from '@/shared'

import { modal, closeModal, open, catchFocus } from '../lib/modal'

const button = createElementRef<HTMLButtonElement>()
```

Но создание переменных для компонента должно быть вынесено в хук этого компонента. Посмотри примеры, как это сделано в любом другом компоненте. Скорее всего, некоторый код из `/lib/modal` надо будет перенести в `/hooks/modal-window`.